lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1338,15 +1386,21 @@ export class UserAgentApplication {
     this._cacheStorage.removeItem(Constants.urlHash);
 
     try {
+      // Clear the cookie in the hash
+      this._cacheStorage.clearCookie();
+      const userState = this.getUserState(this._cacheStorage.getItem(Constants.stateLogin, this.storeAuthStateInCookie));
+      if (error || errorDesc) {
+        if (this._errorReceivedCallback) {
+          this._errorReceivedCallback(new ServerError(error, errorDesc), userState);
+        }
+      }
+      if (token) {
         if (this._tokenReceivedCallback) {
-          // Clear the cookie in the hash
-          this._cacheStorage.clearCookie();
-
           // Trigger callback
           // TODO: Refactor to new callback pattern
-          this._tokenReceivedCallback.call(this, errorDesc, token, error, tokenType,  this.getUserState(this._cacheStorage.getItem(Constants.stateLogin, this.storeAuthStateInCookie)));
+          this._tokenReceivedCallback.call(this, token, tokenType, userState);
```

Comment:
same comment here about .call and applying the this context

Reply:
fixed in future PR